src/Playwright/Helpers/Paths.cs

```diff
@@ -37,7 +37,14 @@ internal static string GetExecutablePath()
             if (!assemblyDirectory.Exists || !File.Exists(Path.Combine(assemblyDirectory.FullName, "Microsoft.Playwright.dll")))
             {
                 var assemblyLocation = typeof(Playwright).Assembly.Location;
-                assemblyDirectory = new FileInfo(assemblyLocation).Directory;
+                if (!string.IsNullOrEmpty(assemblyLocation))
+                {
+                    // For clarification, we only overwrite the directory if the assembly location is
+                    // not empty, like when running from a single file packaged application. In that
+                    // context, getting the dll is unsupported, but also not needed. The rest of the
+                    // functionalities will work, and we can leave it at that.
+                    assemblyDirectory = new FileInfo(assemblyLocation).Directory;
```

Comment:
So `Assembly.Location` is an empty string...  Then you are looking at the `cwd`'s directory? But what is the point of having assemblyDirectory point relative to `cwd`?

Reply:
In cases like LinqPad, the cwd will be different to the assembly location.

Reply:
But will it point anywhere reasonable? Also, this is not the bug you refer to then? What are you fixing?

Reply:
Yes.

So, the default/correct lookup is `AppContext.BaseDirectory`. But, that can point to the wrong place sometimes (like when loaded in LinqPad, or interactive), so we double-check if our DLL is deployed next to it. If it's not, we revert to the assembly location. Now, for special cases, like the one in the last comment (single file apps) of the linked issue, that is *empty*, so we were setting the location to an empty string. 

This particular edge case is that the `.playwright` folder is there, and the path returned from `BaseDirectory` is *NOT EMPTY*, however, it does not have `Microsoft.Playwright.dll` because it's embedded in a single file. So our check incorrectly went down this route.